### PR TITLE
A mish-mash of fixes to allow yk-benchmarks to run.

### PR DESCRIPTION
**Pull request by vext01**
Most notably fix:
 - fix a bug in deoptinfo where optimising them out causes an index skew.
 - Fix (kind of) sign extend of non-byte sized integers.

### CODE REVIEW

File: ykrt/src/compile/jitc_yk/codegen/x64/mod.rs [hunk 1]
```diff
@@ -1569,9 +1596,14 @@ pub(super) struct X64CompiledTrace {
...
     /// The executable code itself.
     buf: ExecutableBuffer,
-    /// Vector of deopt info, tracked here so they can be freed when the compiled trace is
-    /// dropped.
-    deoptinfo: Vec<DeoptInfo>,
+    /// Deoptimisation info: maps a deoptimisation ID to it's information.
+    ///
+    /// Tracked here so they can be freed when the compiled trace is dropped.
+    ///
+    /// FIXME: The key of this shouldn't be a raw integer. I think it should be either GuardIdx or
```
Comment (vext01) [inline ykrt/src/compile/jitc_yk/codegen/x64/mod.rs:1603]: I'll try to fix this next, but in a separate PR.

File: ykrt/src/compile/jitc_yk/aot_ir.rs [hunk 2]
```diff
@@ -1134,8 +1134,8 @@ impl fmt::Display for DisplayableInst<'_> {
         if let Some(instid) = &self.instid {
             if let Some(li) = self.m.line_infos.get(instid) {
                 let path = self.m.path(li.pathidx);
-                // Absolute path: unwrap cannot fail.
-                let basename = path.file_name().unwrap();
+                // Sometimes the filename cannot be determined.
+                let basename = path.file_name().unwrap_or(OsStr::new("<unkonwn-filename>"));
```
Comment (ltratt) [inline ykrt/src/compile/jitc_yk/aot_ir.rs:1138]: "unknown"

**Comment by ltratt** [comment]
Please squash the typo in.

**Comment by vext01** [comment]
Sorry, fixed.

**Comment by ltratt** [comment]
If removing the `assert` succeeds, please squash.

**Comment by vext01** [comment]
Squashed.

(sorry i pushed twice -- forgot to sign the commit)

**Pull request merged**